react-app/src/providers/TransactionProvider.tsx

```diff
@@ -60,17 +63,16 @@ const TransactionProvider: React.FC<TransactionProviderProps> = (props) => {
             success: translate("transaction.success"),
           });
         })
-        .then((tx) => {
-          // TODO: Handle boardcasted state
-          console.log(tx);
+        .then(async () => {
+          return wallet.refreshAccounts();
```

Comment:
is this after waiting for confirmation or simply sent?

Reply:
Submit -> Sign on Keplr -> Close Modal -> Broadcast -> Show broadcasting toast -> Broadcast success -> Show success toast -> Refresh account

Splitted into two stages instead of using `SignAndBroadcast` to prevent the modal from blocking the UI for the entire duration of the transaction broadcast

Reply:
there might be delay b/t broadcast success & inclusion in block?

do we need to wait for a block manually? or the sdk already does that? 🤔

Reply:
oh it handles that already, it will reject the promise if the block is not found on the chain in which i have experienced that when I tried to send a transaction right after restarting skynet

Reply:
ok 👍